fred2/bgbitmapdlg.cpp

```diff
@@ -458,8 +458,8 @@ void bg_bitmap_dlg::OnClose()
 
     The_mission.flags.set(Mission::Mission_Flags::Subspace, m_subspace != 0);
 
-	string_copy(The_mission.skybox_model, m_skybox_model, NAME_LENGTH, 1);
-	string_copy(The_mission.envmap_name, m_envmap, NAME_LENGTH, 1);
+	string_copy(The_mission.skybox_model, m_skybox_model, NAME_LENGTH - 1, 1);
+	string_copy(The_mission.envmap_name, m_envmap, NAME_LENGTH - 1, 1);
```

Comment:
Generally speaking, how did you find the call sites to update? Have all calls to, e.g., `strncpy()` also been checked?

Reply:
I don't remember why I checked `string_copy` but I think it was because it was closely related to replacements in FRED.  IIRC I changed the function name, recompiled, checked all the compilation errors, then changed the name back.  I did not check `strncpy`.